activejob/lib/active_job/enqueuing.rb

```diff
@@ -34,7 +34,60 @@ def perform_later(...)
         enqueue_result
       end
 
+      # Push many jobs onto the queue at once. The parameter +multi_args+ is an array of arrays or hashes.
+      # Each item in the array contains the arguments for a single job. For example, the following arguments
+      # would enqueue two jobs with three arguments each.
+      #
+      # +multi_args+: [[1, "String arg", <#SomeModel ...>], [2, "String arg", <#SomeModel ...>]]
+      # First job enqueued with [1, "String arg", <#SomeModel ...>]
+      # Second job enqueued with [2, "String arg", <#SomeModel ...>]
+      #
+      # For a job with keyword arguments, the arguments can be passed as a hash:
+      #
+      # +multi_args+: [{ foo: "bar", number: 1 }, { foo: "baz", number: 2 }]
+      #
+      # Returns the number of jobs successfully enqueued.
+      def perform_all_later(multi_args, options: {})
+        if queue_adapter.respond_to?(:enqueue_all)
+          jobs = []
+          serialized_jobs = []
+
+          multi_args.each do |args|
+            job = job_from_multi_args(args).set(options)
+            job.run_callbacks(:enqueue, :before) do
+              job.run_callbacks(:enqueue, :around) do
+                jobs << job
+                serialized_jobs << job.serialize
+              end
```

Comment:
It seems potentially confusing that `around_enqueue` callbacks might not wrap the actual enqueuing. For example, Active Job itself currently relies on `around_enqueue` to send instrumentation events. They’ll be inaccurate for jobs enqueued via `perform_all_later`.

It’s also not clear how aborting from a callback should work in this situation (via `throw :abort`). Should it skip enqueuing the one job, or all jobs, or all remaining jobs? The first option seems most desirable to me at the moment, but is fussy to implement.

Would it make sense for `perform_all_later` not to run callbacks, like Active Record’s `insert_all` and `update_all` don’t? That will have the benefit of restoring some of the performance gains of bulk enqueuing, too.